Sources/LanguageServerProtocol/SupportTypes/ServerCapabilities.swift

```diff
@@ -904,18 +1065,48 @@ public struct DiagnosticOptions: WorkDoneProgressOptions, Codable, Hashable {
     self.workDoneProgress = workDoneProgress
   }
 
-  public func encodeIntoLSPAny(dict: inout [String: LSPAny]) {
-    if let identifier = identifier {
+  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+    if case .string(let value) = dictionary["identifier"] {
+      self.identifier = value
+    }
+
+    self.interFileDependencies = false
+    if case .bool(let value) = dictionary["interFileDependencies"] {
+      self.interFileDependencies = value
+    }
+
+    self.workspaceDiagnostics = false
+    if case .bool(let value) = dictionary["workspaceDiagnostics"] {
+      self.workspaceDiagnostics = value
+    }
```

Comment:
Since `interFileDependencies` and `workspaceDiagnostics` are required fields, I think we should return `nil` if they aren’t present. This is what you do for the required fields in the `init(fromLSPDictionary:)` for the other structs as well.